velox/dwio/dwrf/writer/ColumnWriter.cpp

```diff
@@ -2031,6 +2031,12 @@ std::unique_ptr<BaseColumnWriter> BaseColumnWriter::create(
     case TypeKind::BIGINT:
       return std::make_unique<IntegerColumnWriter<int64_t>>(
           context, type, sequence, onRecordPosition);
+    case TypeKind::HUGEINT:
+      if(type.type()->isLongDecimal()) {
```

Comment:
You can add VELOX_CHECK(type.type()->isLongDecimal()) instead of raising an exception here.